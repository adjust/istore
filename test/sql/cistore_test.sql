SELECT 'de::phone::ios=>25,us::phone::ios=>10'::cistore;
SELECT 'de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore;
SELECT cistore('de'::country, 'phone'::device_type, 'ios'::os_name, 10);
SELECT cistore('de'::country, 'phone'::device_type, 'ios'::os_name, 5::smallint, 10::bigint);
SELECT 'de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore + 'de::tablet::ios::10=>25,us::phone::ios::1=>10'::cistore;
-- SELECT agg('de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore, 'country')
-- SELECT agg('de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore, 'device_type')
-- SELECT agg('de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore, 'device_type', 'os_name')
-- SELECT agg('de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore, 'country', 'device_type')
-- SELECT slice('de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore, '*::phone::*')
-- SELECT slice('de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore, 'de::phone::*')
