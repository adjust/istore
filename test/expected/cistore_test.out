SELECT 'de::phone::ios=>25,us::phone::ios=>10'::cistore;
                    cistore                    
-----------------------------------------------
 "us::phone::ios"=>"10","de::phone::ios"=>"25"
(1 row)

SELECT 'de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore;
                       cistore                        
------------------------------------------------------
 "us::phone::ios::1"=>"10","de::phone::ios::10"=>"25"
(1 row)

SELECT cistore('de'::country, 'phone'::device_type, 'ios'::os_name, 10);
        cistore         
------------------------
 "de::phone::ios"=>"10"
(1 row)

SELECT cistore('de'::country, 'phone'::device_type, 'ios'::os_name, 5::smallint, 10::bigint);
          cistore          
---------------------------
 "de::phone::ios::5"=>"10"
(1 row)

SELECT 'de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore + 'de::tablet::ios::10=>25,us::phone::ios::1=>10'::cistore;
                                     ?column?                                     
----------------------------------------------------------------------------------
 "us::phone::ios::1"=>"20","de::phone::ios::10"=>"25","de::tablet::ios::10"=>"25"
(1 row)

-- SELECT agg('de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore, 'country')
-- SELECT agg('de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore, 'device_type')
-- SELECT agg('de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore, 'device_type', 'os_name')
-- SELECT agg('de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore, 'country', 'device_type')
-- SELECT slice('de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore, '*::phone::*')
-- SELECT slice('de::phone::ios::10=>25,us::phone::ios::1=>10'::cistore, 'de::phone::*')
